### PR TITLE
🦄 UNICORN ~*~Liz Baker~*~

### PR DESCRIPTION
**Pull request by LizBaker**
- [x] stock_prices
- [ ]  recipe_batches
- [ ] counting_stairs
- [ ]  rock_paper_scissors
- [ ]  making_change

### CODE REVIEW

File: stock_prices/stock_prices.py [hunk 1]
```diff
@@ -3,7 +3,16 @@
...
-  pass
+  profits = []
+  for i in enumerate(prices):
+    index = i[0]
+    if index < len(prices)-2:
+      for price in prices:
+        if prices.index(price) >= index+1:
+          profits.append(price - prices[index])
+
+  profits.sort(reverse = True)
```
Comment (Mister-Corn) [inline stock_prices/stock_prices.py:14]: Remember, `sort()` is an expensive operation. Can you think of a way where we can avoid it?
(_Hint: We don't care about any profit value except the highest one. Does your program need to remember profit values other than the one we care about?_).

File: stock_prices/stock_prices.py [hunk 2]
```diff
@@ -3,7 +3,16 @@
 import argparse
 
 def find_max_profit(prices):
-  pass
+  profits = []
+  for i in enumerate(prices):
```
Comment (Mister-Corn) [inline stock_prices/stock_prices.py:7]: As you are well aware of, enumerate returns a tuple of `(index, value)`. As an alternative to setting up a new variable to refer to the first element of the tuple, you can destructure right in the for loop:
```
  for index, value in enumerate(prices):
    # index = i[0] Not needed anymore, since index refers directly to the index integer
    if index < len(prices)-2:
       # so on and so forth
```

**Review by Mister-Corn — COMMENTED** [review]
Good start, Liz. Lemme know if you need any help, as we should have this repo done before 3:30 PM PDT 😱

File: recipe_batches/recipe_batches.py [hunk 3]
```diff
@@ -2,13 +2,31 @@
...
+  pendingBatch = []
+  for ingredient in recipe:
+    if ingredient in ingredients: #check if it has ingredient at all
+      if recipe.get(ingredient) <= ingredients.get(ingredient): #check if it has enough ingredient
+        pendingBatch.append(1) #if so, add a tally to pending
+        ingredients[ingredient] -= recipe.get(ingredient) #decrement from ingredients
+        if len(pendingBatch) == len(recipe): #check if the pending batch is finished
+          batch += 1 #if it is, increment batch
+          pendingBatch = [] #and reset pending
+          recipe_batches(recipe, ingredients, batch) #try again to make another batch
```
Comment (Mister-Corn) [inline recipe_batches/recipe_batches.py:16]: I think I see why you're getting `None`. In order to get something out of a function, recursively called or not, you need a `return`. In line 16, you are making a recursive call to `recipe_batches()`. Ultimately, your base cases will return either `batch` or `0`, but without a return here, you will never see it. Your base case execution contexts will resolve, then resolve upwards until it hits here, but this execution context will never cough up the output without a return.

TLDR: Put a `return` before the recursive function call on line 16.
